Load/Save Modal: Added Link to Help Page

microbit-henry opened the pull request:
The Microbit Program Files section of the Load/Save Modal was not clear, so I've added a link to /help.html#fs.

Addresses microbit-foundation/platform-software-issue-tracker#471

microbit-carlos commented:
Great work Henry!

Can we try a small message, and maybe icon, maybe something like this:

![image](https://user-images.githubusercontent.com/29712657/65341735-ed23cf80-dbc8-11e9-9dd1-1b06d0f1ce50.png)

I used this fontawesome icon: https://fontawesome.com/v4.7.0/icon/info-circle

Another thing is that the space between the download buttons and the "files section" title has been reduced.

From:

![image](https://user-images.githubusercontent.com/29712657/65341797-180e2380-dbc9-11e9-9e26-baff6a5cd053.png)

To:

![image](https://user-images.githubusercontent.com/29712657/65341789-12b0d900-dbc9-11e9-9593-f7e86b6ab6db.png)

Can we restore the extra whitespace?

microbit-henry commented:
<img width="833" alt="Screenshot 2019-09-20 17 38 15" src="https://user-images.githubusercontent.com/53710678/65343747-ae444880-dbcd-11e9-9e98-d1d064e3f675.png">
I've tried to match the white space with the white space after the other titles in the modal, do you want it to be bigger @microbit-carlos? Also, I've set the File Help text to bold, but not sure if it looks good. Do you prefer it without?
<img width="843" alt="Screenshot 2019-09-20 17 39 11" src="https://user-images.githubusercontent.com/53710678/65343828-e77cb880-dbcd-11e9-9bbe-81bbd89626e4.png">

microbit-carlos commented:
I think the boldness of the icon is good enough, so we can keep the text without it.

Can we add a bit more white space at the top of "Microbit Program Files"? Also we should match the whitespace on top of the "Save" title, that way the three sections of the modal are more clearly separated.

microbit-mark commented:
Not sure on this, but I wondered if it was more conventiaonal to have the icon directly after with the text that it's referencing eg
![65341735-ed23cf80-dbc8-11e9-9dd1-1b06d0f1ce50](https://user-images.githubusercontent.com/31242877/65408896-5cbed800-ddde-11e9-91c8-a118129301d9.png)
and also if we're linking to the help file, would a '?' be more appropriate as it's already a familiar icon in the UI and it's more than a tooltip. There's discussion about this here
https://ux.stackexchange.com/questions/64920/info-icon-vs-question-mark

microbit-carlos commented:
While I do agree that the icon at the end is more conventional, I think we need to highlight this help as much as possible, and I don't think the small icon on its own catches enough attention from the user.

One of the reasons I I think the info icon might work slightly better than the question mark, is because having the icon followed by the text looked a bit weird, like opening the sentence with a question mark: `? Files help`

@microbit-henry Could you screenshot what the modal would look like with the question mark icon instead?

microbit-henry commented:
@microbit-carlos this is what it looks like with a questions mark. I've matched the spaces above and below the 'Microbit Program Files' with the other headers.
<img width="833" alt="Screenshot 2019-09-23 15 11 26" src="https://user-images.githubusercontent.com/53710678/65433308-aaeece00-de14-11e9-883b-2e9df0553640.png">

microbit-henry commented:
Just added some more space below the headers, does this look any better? @microbit-carlos 
<img width="830" alt="Screenshot 2019-09-23 15 22 13" src="https://user-images.githubusercontent.com/53710678/65433997-f786d900-de15-11e9-8032-385417a07d1a.png">

Regarding the help link, I have switched it from resting on the same line as the header to aligning its centre with the headers centre. Do you have a preference? 
<img width="727" alt="Screenshot 2019-09-23 15 25 43" src="https://user-images.githubusercontent.com/53710678/65434247-6a904f80-de16-11e9-8b1c-0a1c53745c14.png">

microbit-mark commented:
I think as it's a new feature and '<name of program> Files' isn't descriptive, the '?' feels and looks right at this stage. 

My only other thought for this is to do what I think we describe in the ticket as 
" -  Short paragraph at the files section?"

eg keep the 'i' but rather than link to help, on click/hover reveal/unhide a sentence underneath the title to describe what the filesystem is and then link off to the help from that, that way we give some contextual support without the user leaving the editor or modal window.

microbit-henry commented:
@microbit-mark I added one sentence with a link as a placeholder. Does this look good to you, and do you have a more suitable suggestion for the text?
<img width="824" alt="Screenshot 2019-09-24 11 18 45" src="https://user-images.githubusercontent.com/53710678/65503620-32901780-debd-11e9-8036-51c303ed5540.png">

microbit-mark commented:
I like this, as it gives some contextual help. I would avoid using 'Microbit Program Files' as that changes if the user changes the program title and avoid using 'click here' as it isn't a descriptive link for accessibility eg  screenreader, so something like:

"The [Python Editor Files](python.microbit.org/beta/help.html#fs) area shows you the files included in your program and lets you add or remove external python modules and other files."

microbit-carlos commented:
I'm not completely sold on the text (yet!), I'll spend some time looking into this tonight.

But in the mean time, could we change the `{{project_name}} Files` (by default project_name is `microbit program`) to "Project Files"? It works better and I think it also goes well with the "Download Project Hex" button.

microbit-henry commented:
> I'm not completely sold on having a block of text (yet!), I'll spend some time looking into this tonight.
> 
> But in the mean time, could we change the `{{project_name}} Files` (by default project_name is `microbit program`) to "Project Files"? It works better and I think it also goes well with the "Download Project Hex" button.

Do you mean like this? or still using the braces?
<img width="691" alt="Screenshot 2019-09-25 14 44 56" src="https://user-images.githubusercontent.com/53710678/65606636-2bdbd000-dfa3-11e9-9ca4-11af316d2d74.png">

microbit-carlos commented:
> Do you mean like this? or still using the braces?
> <img alt="Screenshot 2019-09-25 14 44 56" width="691" src="https://user-images.githubusercontent.com/53710678/65606636-2bdbd000-dfa3-11e9-9ca4-11af316d2d74.png">

Yes.

microbit-carlos commented on editor.html:
```diff
@@ -92,7 +92,13 @@ <h2 class="modal-title"><i class="fa fa-download"></i> <strong>{{ save-title }}<
-            <h2 class="modal-title"><i class="fa fa-download"></i> <strong>{{ program-title }} {{ fs-title }}</strong></h2>
+            <div id="addFile">
+              <div id="addFileHeader">
+                <h2 class="modal-title"><i class="fa fa-download"></i> <strong>{{ help-button }}</strong></h2>
```
This is the files title, why is it called `help-button`?

microbit-carlos commented on editor.html:
```diff
@@ -92,7 +92,13 @@ <h2 class="modal-title"><i class="fa fa-download"></i> <strong>{{ save-title }}<
+              <div id="addFileHeader">
+                <h2 class="modal-title"><i class="fa fa-download"></i> <strong>{{ help-button }}</strong></h2>
+              </div>
+              <div id="addFileHelp"><a id="expandHelpPara"><i class="fa fa-info-circle" aria-hidden="true"> <span>Files Help</span></i></a></div>
```
`Files Help` also needs translation.

microbit-carlos commented on editor.html:
```diff
@@ -92,7 +92,13 @@ <h2 class="modal-title"><i class="fa fa-download"></i> <strong>{{ save-title }}<
+              </div>
+              <div id="addFileHelp"><a id="expandHelpPara"><i class="fa fa-info-circle" aria-hidden="true"> <span>Files Help</span></i></a></div>
+            </div>
+            <div id="fileHelpPara">{{ file-help-text }}<a href=\'help.html#fs\' target="_blank">{{ help-link }}</a></div>
```
```suggestion
            <div id="fileHelpPara">{{ file-help-text }}<a href="help.html#fs" target="_blank">{{ help-link }}</a>.</div>
```

microbit-carlos commented on lang/en.js:
```diff
@@ -118,7 +118,11 @@ var language = {
+    'help-button' : 'Project Files',
+    'file-help-text' : 'The Project Files area shows you the files included in your program and lets you add or remove external python modules and other files. Find out more in the ',
+    'help-link' : 'Python Editor help documentation'
+
```
Don't forget to add these entries into the other language files as well.

microbit-carlos commented on editor.html:
```diff
@@ -87,12 +87,18 @@ <h2 class="modal-title"><i class="fa fa-upload"></i> <strong>{{ load-title }}</s
+              </div>
+              <div id="addFileHelp"><a id="expandHelpPara"><i class="fa fa-info-circle" aria-hidden="true"> <span>{{ help-button }}</span></i></a></div>
+            </div>
+            <div id="fileHelpPara">{{ file-help-text }}<a href='help.html#fs' target="_blank">{{ help-link }}</a></div>
```
The HTML attributes should use double quotes.

```suggestion
            <div id="fileHelpPara">{{ file-help-text }}<a href="help.html#fs" target="_blank">{{ help-link }}</a>.</div>
```

Also, we should have a period at the end of the sentence.
